db/indexes.go

```diff
@@ -458,3 +459,15 @@ func replaceSyncTokensQuery(statement string, useXattrs bool) string {
 func replaceIndexTokensQuery(statement string, idx SGIndex, useXattrs bool) string {
 	return strings.Replace(statement, indexToken, idx.fullIndexName(useXattrs), -1)
 }
+
+// Replace $activeOnlyFilter placeholder with activeOnlyFilterFilterExpression if activeOnly is true and the limit
+// is grater than zero. If activeOnly is false or limit is not grater than 0, the $activeOnlyFilter placeholder will
+// be replaced with an empty string in the channel query statement.
+func replaceActiveOnlyFilter(statement string, limit int, activeOnly bool) string {
+	activeOnlyFilterFilterExpression := "($sync.flags IS MISSING OR BITTEST($sync.flags,1) = false) AND"
+	if limit > 0 && activeOnly {
```

Comment:
Is there a reason not to use the filter expression when limit=0 and activeOnly=true?

Reply:
No. Actually we can ignore the limit check here, only need to check for activeOnly=true before applying the filter expression.